### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by ZNielsen**
## Check list

- [x] I have read through the [README](https://github.com/wfxr/minimap.vim/blob/master/README.md) (especially F.A.Q section)
- [x] I have searched through the existing issues or pull requests
- [x] I have performed a self-review of my code and commented hard-to-understand areas
- [x] I have made corresponding changes to the documentation (when necessary)

## Description

Adds unit tests. See the README for how to run. 

I am not adding the automated checks into this, I need more time to troubleshoot it. Putting this up because it probably won't change and it provides some benefit while I figure out the GitHub workflow stuff.

### Details on code changes
Changes in vim.vim are mostly relocating git/search specific functions so they are together.  I also took out the multiple parameter passing and replaced it with passing in the whole window info table (tables are passed by reference, so there is no performance hit). It cleans up the interfaces a lot and lets us reuse information instead of recalculating.

## Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] Improvement of existing features
- [x] Refactor
- [ ] Breaking change
- [x] Documentation change
- [x] CI / CD

## Test environment

- OS
    - [ ] Linux
    - [x] Mac OS X
    - [ ] Windows
    - [ ] Others:
- Vim
    - [x] Neovim: 0.5.0
    - [ ] Vim: <Version>

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 - [ ] I have read through the [README](https://github.com/wfxr/minimap.vim/blob/master/README.md) (especially F.A.Q section)
 - [ ] I have searched through the existing issues or pull requests
+- [ ] I have verified all existing unit tests pass (See the README on how to run)
```
Comment (ZNielsen) [inline .github/pull_request_template.md:7]: This check can be removed once we automate the testing.

**Comment by wfxr** [comment]
@ZNielsen Thank you for your excellent work! I'm a little busy lately and have no enough time for a detailed review. I did a rough test locally and everything works fine.

**Review by wfxr — APPROVED** [review]

File: .github/pull_request_template.md (hunk 1 above)
Comment (wfxr) [reply .github/pull_request_template.md:7]: Adding the following steps into the ci workflow seems to be a feasible solution?

```sh
mkdir -p ~/.vim/autoload
curl -fLo ~/.vim/autoload/plug.vim https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim
# generate a minimal `vimrc` which loads `vim-testify` and `minimap.vim`
vim +PlugInstall +qall
vim +TestifySuite
```

**Comment by ZNielsen** [comment]
No worries, open source moves at its own pace.

**Pull request merged**